docs/Getting-Started.md

```diff
@@ -56,61 +44,167 @@ fastify.listen(3000, function (err) {
+
+We have used the `decorate` api. Let's take a moment to understand what is and how it works. A typic thing that you'll need is to use the same code/library in different parts of your application. A solution is to require the code/library that you need everywhere, it works, but is annoying because of code repeated and if needed, long refactors.  
+To solve this Fastify offers you a nice api, `decorate`, which add your code/library in the Fastify namespace, in way that you can use it everywhere.
+
+If you want to get deepen on how plugins works, how can you develop plugins for Fastify or use all the apis that the framework offers to share your code around the application and how handle the complexity given by the asynchronous bootstrapping, checkout [the hitchhiker's guide to plugins](https://github.com/fastify/fastify/blob/master/docs/Plugins.md).
+
+<a name="validate-data"></a>
+### Validate your data
+Data validation is extremely important and is a core concept of the framework.  
+To validate your incoming requests, Fastify uses [JSON Schema](http://json-schema.org/).
+Let's see an example on how validate your routes:
+```js
+const opts = {
+  schema: {
+    body: {
+      type: 'object',
+      properties: {
+        someKey: { type: 'string' },
+        someOtherKey: { type: 'number' }
+      }
+    }
+  }
+}
+
+fastify.post('/', opts, async (request, reply) => {
+  return { hello: 'world' }
+})
+```
+As you can see you can pass an options object to the route, which accepts a `schema` key, that will contain all the schemas for your route, `body`, `querystring`, `params` and `headers`.  
+Checkout [Validation and Serialization](https://github.com/fastify/fastify/blob/master/docs/Validation-and-Serialization.md) to know more.
+
+<a name="serialize-data"></a>
+### Serialize your data
+Fastify comes with a first class support for JSON. It is extremely optimized to parse JSON body and to serialize JSON output.  
+To speed up your JSON serialization (yes, it is slow!) you must use the `response` key of the schema option. Following hot it works:
+```js
+const opts = {
+  schema: {
+    response: {
+      200: {
+        type: 'object',
+        properties: {
+          hello: { type: 'string' }
+        }
+      }
+    }
+  }
+}
+
+fastify.get('/', opts, async (request, reply) => {
+  return { hello: 'world' }
+})
+```
+Just by using the above code you will speed up your serialization by 2x or even 3x than before, and you will be also protected from the leaking of sensitive data, since the Fastify will serialize just the data present in your response schema.  
+Checkout [Validation and Serialization](https://github.com/fastify/fastify/blob/master/docs/Validation-and-Serialization.md) to know more.
+
+<a name="extend-server"></a>
+### Extend your server
+Fastify is built to be extremely extensible and it's very minimal, we believe that provide a fully featured framework is not needed, since it will probably give you more than what you need.  
```

Comment:
> Fastify is built to be extremely extensible and ~it's~ very minimal. ~we believe that provide a fully featured framework is not needed, since it will probably give you more than what you need.~ We believe that a bare minimum framework is all that is necessary to make great applications possible.